platforms/software/dependency-management/src/main/java/org/gradle/internal/component/local/model/TransformedComponentFileArtifactIdentifier.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.component.local.model;
+
+import org.gradle.api.artifacts.component.ComponentArtifactIdentifier;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.internal.DisplayName;
+
+import java.util.Objects;
+
+/**
+ * Identifies the transformed artifact of a component. The original file name is tracked in order to guarantee uniqueness,
+ * as artifact transformations may result in multiple artifacts with the same file name.
+ */
+public class TransformedComponentFileArtifactIdentifier implements ComponentArtifactIdentifier, DisplayName {
```

Comment:
❓ Does this have any effects on Build Scans? Do they currently capture `ComponentArtifactIdentifier`s of some kind or only `ComponentIdentifer`s?

Reply:
You'd see this from a tooling API consumer, though, like in IDEA.

Reply:
I think the returned display name is still reasonable. It is is a display name so it should not drive production logic